content/articles/http-requests-with-ktor-client-in-android-jetpack-compose/index.md

```diff
@@ -0,0 +1,352 @@
+    producer = {
+        value = apiService.getProducts()
+        }
+)
+```
+
+Let's now set up the `LazyColumn`. We will do this inside the `Surface{}`.
+
+```kotlin
+LazyColumn {
+    items(products.value) {
+        Box(
+            modifier = Modifier
+                .fillMaxWidth()
+                .padding(
+                    bottom = 6.dp,
+                    top = 6.dp,
+                )
+                .background(Color.Gray)) {
+
+            Column(
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .padding(4.dp),
+                horizontalAlignment = Alignment.CenterHorizontally,
+            ) {
+                //set the image url
+                val painter = rememberImagePainter(
+                    data = it.image,
+                    builder = {
+                        error(R.drawable.ic_launcher_background)
+                    }
+                )
+
+                Image(
+                    modifier = Modifier
+                        .fillMaxWidth()
+                        .height(150.dp),
+                    contentScale = ContentScale.Crop,
+                    contentDescription = "Coil Image",
+                    painter = painter
+                )
+                Spacer(
+                    modifier = Modifier
+                        .height(4.dp)
+                )
+                Text(
+                    text = it.title,
+                    fontSize = 18.sp
+                )
+                Spacer(
+                    modifier = Modifier
+                        .height(4.dp)
+                )
+                Text (
+                    text = it.description,
+                    fontSize = 12.sp
+                )
+            }
+        }
+    }
+}
+```
+
+This column represents an actual Recycler View. Here we are using the Jetpack Compose to populate the data to a cleaned and minimalist UI with less code.
+
+We are wrapping each item in a `Box` that has three properties, the `Image` and two `Text` blocks.
+
+First, we will process the returned data and get the value of the `image`. The returned string is a URL that loads an image from the server. To load these images, we are using a Coil image loader. Here we set the image URL, pass the parameter `data` and assign it the value of the returned image URL. This will launch an image request. If return `true`, we proceed with the request and load the image into the Image view using the Painter. This will create a composable that lays out and draws a given Painter. This will attempt to size the composable according to the Painter's intrinsic size and the `Modifier` parameters.
+
+If it returns `false`, we skip executing the request and builder an optional lambda) configure the request, and set an error drawable resources. The Text composable will load the values of product `title` and `description`.
+
+The app is now ready, and you can run it to test if everything works as expected.
+
```

Comment:
Just before the conclusion, provide at least one image - a screenshot of what to expect upon running the app.